tests/helpers/test_supplier_data_helpers.py

```diff
@@ -2,6 +2,32 @@
 import requests
 
 from dmscripts.helpers.supplier_data_helpers import country_code_to_name
+from dmscripts.data_retention_remove_supplier_declarations import SupplierFrameworkDeclarations
+from tests.assessment_helpers import BaseAssessmentTest
+import mock
+from dmapiclient import DataAPIClient
+import json
+
+
+class TestSupplierFrameworkDeclarations(BaseAssessmentTest):
+
+    def _supplier_framework_response(self):
+        with open("tests/fixtures/test_supplier_frameworks_response.json", 'r') as response_file:
+            return json.load(response_file)['supplierFrameworks']
+
+    def test_suppliers_application_failed_to_framework(self):
+        mocked_api_client = mock.create_autospec(DataAPIClient)
+        mocked_api_client.find_framework_suppliers_iter.side_effect = lambda *args, **kwargs: \
+            iter(self._supplier_framework_response())
+        supplier_framework = SupplierFrameworkDeclarations(mocked_api_client, mock.ANY, 'g-cloud-8', dry_run=False)
+        assert supplier_framework.suppliers_application_failed_to_framework() == [12345, 23456]
```

Comment:
One more assert on the `mocked_api_client. remove_supplier_declaration ` call would be ideal here, but we can add that in when we do the failed applications scenario.